File: library/src/main/java/info/hannes/cvscanner/DocumentTrackerFactory.java
package info.hannes.cvscanner;

import com.google.android.gms.vision.MultiProcessor;
import com.google.android.gms.vision.Tracker;

import online.devliving.mobilevisionpipeline.GraphicOverlay;


public class DocumentTrackerFactory implements MultiProcessor.Factory<Document> {
    GraphicOverlay<DocumentGraphic> mOverlay;
    DocumentTracker.DocumentDetectionListener mListener;

    public DocumentTrackerFactory(GraphicOverlay<DocumentGraphic> mOverlay, DocumentTracker.DocumentDetectionListener mListener) {
        this.mOverlay = mOverlay;
        this.mListener = mListener;
    }

    @Override
    public Tracker<Document> create(Document document) {
        DocumentGraphic graphic = new DocumentGraphic(mOverlay, document);
        return new DocumentTracker(mOverlay, graphic, mListener);
    }
}

File: library/src/main/java/info/hannes/cvscanner/DocumentTrackerFactory.kt
package info.hannes.cvscanner

import com.google.android.gms.vision.MultiProcessor
import com.google.android.gms.vision.Tracker
import info.hannes.cvscanner.DocumentTracker.DocumentDetectionListener
import info.hannes.visionpipeline.GraphicOverlay

class DocumentTrackerFactory(private val graphicOverlay: GraphicOverlay<DocumentGraphic>, private val documentDetectionListener: DocumentDetectionListener) : MultiProcessor.Factory<Document> {

    override fun create(document: Document): Tracker<Document> {
        val graphic = DocumentGraphic(graphicOverlay, document)
        return DocumentTracker(graphicOverlay, graphic, documentDetectionListener)
    }

}